pkg/engine/provider/provider.go

```diff
@@ -12,13 +12,13 @@ import (
 type Sink func(ctx context.Context, filename string, content io.ReadCloser) error
 
 // ResolverSink defines a sink function to be passed as reference to functions for resolved files/templates
-type ResolverSink func(ctx context.Context, filename string) error
+type ResolverSink func(ctx context.Context, filename string) ([]string, error)
 
 // SourceProvider is the interface that wraps the basic GetSources method.
 // GetBasePath returns base path of FileSystemSourceProvider
 // GetSources receives context, receive ID, extensions supported and a sink function to save sources
 type SourceProvider interface {
 	GetBasePath() string
 	GetSources(ctx context.Context, extensions model.Extensions, sink Sink, resolverSink ResolverSink) error
-	checkConditions(info os.FileInfo, extensions model.Extensions, path string) (bool, error)
+	checkConditions(info os.FileInfo, extensions model.Extensions, path string, resolved bool) (bool, error)
```

Comment:
Couldn't this be removed from interface definition? Since it is a private function I don't see any reason to this be part of interface definition...

Reply:
I agree with removing this private function from the interface.